frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts

```diff
@@ -416,4 +471,77 @@ export class WorkPackageTimelineTableController extends UntilDestroyedMixin impl
       }
     }
   }
+
+  manageCollapsedGroupHeaderCells(allGroups:GroupObject[],
+                                  collapsedGroupsChange:{[key:string]:boolean}|undefined,
+                                  tableWorkPackages:WorkPackageResource[],
+                                  collapsedGroupsCellsMap:IGroupCellsMap,
+                                  updateAllHeaderCells?:boolean) {
+    if (!collapsedGroupsChange) {
+      return;
+    }
+
+    const collapsedGroupChangesToManage = Object.keys(collapsedGroupsChange).filter(groupKey => {
+      const keyGroupType = groupKey.split('-')[0];
+
+      return this.groupTypesWithHeaderCellsWhenCollapsed.includes(keyGroupType);
```

Comment:
I do not think that it actually necessary to check this for every group. You could also check if the value returned by passing any group, e.g. only the first to https://github.com/opf/openproject/blob/dev/frontend/src/app/components/wp-fast-table/builders/modes/grouped/grouped-rows-helpers.ts#L27